Sources/Build/BuildPlan.swift

```diff
@@ -1833,53 +1850,64 @@ public class BuildPlan {
-            guard let library = info.libraries.first(where: {
-                return $0.platform == buildParameters.triple.os.asXCFrameworkPlatformString && $0.architectures.contains(buildParameters.triple.arch.rawValue)
+            guard let library = metadata.libraries.first(where: {
+                $0.platform == buildParameters.triple.os.asXCFrameworkPlatformString && $0.architectures.contains(buildParameters.triple.arch.rawValue)
             }) else {
-                diagnostics.emit(error: """
+                throw StringError("""
                     artifact '\(target.name)' does not support the target platform and architecture \
                     ('\(buildParameters.triple)')
                     """)
-                return nil
             }
 
             let libraryDirectory = target.artifactPath.appending(component: library.libraryIdentifier)
             let binaryPath = libraryDirectory.appending(component: library.libraryPath)
             let headersPath = library.headersPath.map({ libraryDirectory.appending(component: $0) })
+
             return LibraryInfo(binaryPath: binaryPath, headersPath: headersPath)
         }
+    }
 
-        // If we don't have the library information yet, calculate it.
-        if let xcFramework = xcFrameworkCache[target] {
-            return xcFramework
-        }
+    /// Extracts the executables info from an executablesArchive
+    private func parseToolsArchive(for target: BinaryTarget) throws -> [ToolInfo] {
+        try self.externalToolsCache.memoize(key: target) {
+            let metadata = try ToolsArchiveMetadata.parse(fileSystem: self.fileSystem, rootPath: target.artifactPath)
 
-        let xcFramework = calculateLibraryInfo()
-        xcFrameworkCache[target] = xcFramework
-        return xcFramework
+            // filter the tools that are relevant to the triple
+            let supportedTools = metadata.tools.filter { $0.value.contains(where: { $0.supportedTriplets.contains(buildParameters.triple) }) }
+            // flatten the tools for each access
+            return supportedTools.reduce(into: [ToolInfo](), { partial, entry in
+                let tools = entry.value.map {
+                    ToolInfo(name: entry.key, binaryPath: target.artifactPath.appending(RelativePath($0.path)))
+                }
+                partial.append(contentsOf: tools)
+            })
+        }
     }
 }
 
-/// Information about a library.
+/// Information about a library from a binary dependency.
 private struct LibraryInfo: Equatable {
-
     /// The path to the binary.
     let binaryPath: AbsolutePath
 
     /// The path to the headers directory, if one exists.
     let headersPath: AbsolutePath?
 }
 
+/// Information about an executable from a binary dependency.
+private struct ToolInfo: Equatable {
+    /// The tool name
+    let name: String
+    /// The path to the binary.
+    let binaryPath: AbsolutePath
```

Comment:
This could be where we add the support-files directory if we add one?

Reply:
exactly

Reply:
Per the previous comment, we might want to leave that out for now and have the extension form the path to support files relative to the path of the tool.

Reply:
So if this path is the path to the actual executable then maybe it should be `toolPath` or `execPath` since the term "binary" might be ambiguous?

Reply:
Hmm, although the library kind also calls it a binary, so I guess we should either call it a binary for both or call each one what it is.  Maybe "binary" isn't confusing here, but it seemed as if we use that for the target as a whole plus for the individual library or executable in the binary target.

Reply:
changes to `executablePath`